pkg/e2e/infra/infra.go

```diff
@@ -31,84 +29,68 @@ var _ = g.Describe("[Feature:Machines] Managed cluster should", func() {
 		o.Expect(err).NotTo(o.HaveOccurred())
 	})
 
-	g.It("have ability to additively reconcile taints", func() {
+	g.It("have ability to additively reconcile taints from machine to nodes", func() {
 		var err error
 		client, err := e2e.LoadClient()
 		o.Expect(err).NotTo(o.HaveOccurred())
 
-		g.By("Verify machine taints are getting applied to node")
-		err = func() error {
-			listOptions := runtimeclient.ListOptions{
-				Namespace: e2e.TestContext.MachineApiNamespace,
-			}
-			machineList := mapiv1beta1.MachineList{}
+		machines, err := getMachines(client)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		machine := machines[0]
```

Comment:
Is this always guaranteed to have 1 machine? We may not error, but equally `len(machines)` may be zero.

Reply:
It's guaranty https://github.com/openshift/cluster-api-actuator-pkg/pull/44/files#diff-a8f8018153bd2312214f96573fe48e57R166

Reply:
> It's guaranteed https://github.com/openshift/cluster-api-actuator-pkg/pull/44/files#diff-a8f8018153bd2312214f96573fe48e57R166

Then I think it is worth adding an assertion before accessing `machine[0]`. It makes the intent clearer (for new readers of the code). At this point in the test I know there is at least 1 machine.

Reply:
It should be up to consumer of `getMachines` to decide if zero number of machines is valid or invalid.